### PR TITLE
fix: HEAD route search

### PR DESCRIPTION
**Pull request by ivan-tymoshenko**
Fixes https://github.com/fastify/fastify/issues/4866
PR in find-my-way: https://github.com/delvedor/find-my-way/pull/337

### CODE REVIEW

**Comment by ivan-tymoshenko** [comment]
should be ready to merge

File: lib/route.js [hunk 1]
```diff
@@ -322,7 +322,7 @@ function buildRouting (options) {
         constraints.version = opts.version
       }
 
-      const headHandler = router.find('HEAD', opts.url, constraints)
+      const headHandler = router.findRoute('HEAD', opts.url, constraints)
       const hasHEADHandler = headHandler != null
```
Comment (gurgunday) [inline lib/route.js:326]: nit: if findRoute can’t return undefined, can you make this a strict check

```suggestion
      const hasHEADHandler = headHandler !== null
```

**Review by mcollina — APPROVED** [review]
lgtm

File: test/constrained-routes.test.js [hunk 2]
```diff
@@ -774,3 +860,41 @@ test('error in async constraints', async (t) => {
     t.equal(statusCode, 500)
   }
 })
+
+test('Allow regex constraints in routes', t => {
```
Comment (Eomm) [inline test/constrained-routes.test.js:864]: Should we test the head route here?

**Review by Eomm — APPROVED** [review]
Just a doubt 👍🏼

**Review by metcoder95 — APPROVED** [review]

File: test/constrained-routes.test.js (hunk 2 above)
Comment (Rantoledo) [reply test/constrained-routes.test.js:864]: Yes, that is correct.

**Pull request merged**